dohproxy/httpproxy.py

```diff
@@ -47,6 +47,11 @@ async def doh1handler(request):
             return aiohttp.web.Response(status=400, body=e.body())
     elif request.method == 'POST':
         body = await request.content.read()
+        # size without length indicator
+        if len(body) - 2 > constants.DOH_MAX_QUERY_SIZE:
```

Comment:
There is no length indicator when sent as part of HTTP payload. This is also confirmed on line 64 (and 68 :)) where the message is parsed directly from body.

Reply:
Good catch :+1: